server/build_redirect.js

```diff
@@ -1,13 +1,20 @@
 import { extract, parse, stringify } from "query-string";
 
-const redirect = ({ organization, long_url, query = {}, referrer = "" }) => {
+const redirect = ({ ship, organization, long_url, query = {}, referrer = "" }) => {
+  const [namespace] = organization.split(".");
```

Comment:
You need to keep the top.domain here to eventually build the URL to redirect to.... Please do not hardcode hullapp.io there